File: Python/Permutation in String.py
class Solution(object):
    def checkInclusion(self, s1, s2):
        if len(s1)>len(s2):
            return False
        sub_count=[0]*26
        window_count=[0]*26

        for i in range (len(s1)):
            sub_count[ord(s1[i])-ord("a")]+=1
            window_count[ord(s2[i])-ord("a")]+=1

        if sub_count==window_count:
            return True

        for j in range(len(s1),len(s2)):
            window_count[ord(s2[j])-ord("a")]+=1
            window_count[ord(s2[j-len(s1)])-ord("a")]-=1
            if sub_count==window_count:
                return True

        return False

File: Java/Permutation-in-string.java
class Solution {
    public boolean checkInclusion(String s1, String s2) {
        int[] window = new int[26];
        int[] s1feq = new int[26];

        for(char ch : s1.toCharArray()){
            s1feq[ch-'a']++;
        }
        int i =0;
        for(int j = 0 ; j < s2.length() ; j++){
            window[s2.charAt(j)-'a']++;

            if(j-i+1>s1.length()){
                window[s2.charAt(i)-'a']--;
                i++;

            }
            if(Arrays.equals(window,s1feq)) return true;

        }
        return false;
        
    }
}
